GitUI/AutoCompletion/CommitMessageMetadataProvider.cs

```diff
@@ -0,0 +1,19 @@
+﻿namespace GitUI.AutoCompletion;
+
+internal class CommitMessageMetadataProvider : IAutoCompleteProvider
+{
+    private static readonly string[] keywords = [
+        "Co-authored-by: ",
+        "Signed-off-by: ",
+        "BREAKING CHANGE: ",
+        "Reviewed-by: ",
+        "Tested-by: ",
+        ];
+
+    private static readonly IEnumerable<AutoCompleteWord> _autoCompleteWords = keywords.Select(k => new AutoCompleteWord(k)).ToArray();
```

Comment:
According to an article in the German dotnetpro, the performance of LINQ has drastically been improved in .NET 8.
But using `IEnumerable<T>` for an array or `List<T>` still can have significant speed and allocation penalties due to boxing in some situations, which is avoided by the optimizer in `foreach` both with and without LINQ expressions.
`for` loops with index operator seem to run similarly slow as `IEnumerable<T>`.

Just to share this info. I am not requesting a change of the interface `IAutoCompleteProvider`.

Reply:
I totally missed it (thanks @mstv). For the private members which aren't directly exposed outside the containing type it's recommended to use concrete types (such as `AutoCompleteWord[]` or `List<AutoCompleteWord>`) instead of abstractions (such as `IEnumerable<AutoCompleteWord>`).

Reply:
Here, no problem of performance because that's done only once.
I used `IEnumerable` only because the `Task.FromResult()` was complaining but I have used `.AsEnumerable()` and it is ok now...